pyboreas/utils/odometry.py

```diff
@@ -386,16 +387,49 @@ def get_path_from_Tvi_list(T_vi_odom, T_vi_gt):
         path_odom (np.ndarray): K x 3 numpy array of estimated xyz coordinates
         path_gt (np.ndarray): K x 3 numpy array of groundtruth xyz coordinates
     """
-
     assert len(T_vi_odom) == len(T_vi_gt)  # assume 1:1 correspondence
     T_iv_odom = [np.linalg.inv(T_vk_i_odom) for T_vk_i_odom in T_vi_odom]
-
     T_iv_gt = [np.linalg.inv(T_vk_i_gt) for T_vk_i_gt in T_vi_gt]
-    T_odom_gt_i = T_iv_odom[0] @ np.linalg.inv(T_iv_gt[0])  # align the first pose
-    T_iv_gt_aligned = [T_odom_gt_i @ T_i_vk_gt for T_i_vk_gt in T_iv_gt]
 
-    path_odom = np.array([T_i_vk[:3, 3] for T_i_vk in T_iv_odom], dtype=np.float64)
-    path_gt = np.array([T_i_vk[:3, 3] for T_i_vk in T_iv_gt_aligned], dtype=np.float64)
+    # Zero out T_iv_gt so first pose is 0
+    T_iv_gt_0 = T_iv_gt[0]
+    T_iv_gt = [np.linalg.inv(T_iv_gt_0) @ T_iv_gt_i for T_iv_gt_i in T_iv_gt]
+
+    # This code aligns the first pose of the groundtruth with the first pose of the estimated trajectory
+    # T_odom_gt_i = T_iv_odom[0] @ np.linalg.inv(T_iv_gt[0])  # align the first pose
+    # T_iv_gt_aligned = [T_odom_gt_i @ T_i_vk_gt for T_i_vk_gt in T_iv_gt]
+
+    # path_odom = np.array([T_i_vk[:3, 3] for T_i_vk in T_iv_odom], dtype=np.float64)
+    # path_gt = np.array([T_i_vk[:3, 3] for T_i_vk in T_iv_gt_aligned], dtype=np.float64)
+
+    #This code aligns the first pose of the estimated trajectory with the first pose of the groundtruth
+    T_gt_odom_i = T_iv_gt[0] @ np.linalg.inv(T_iv_odom[0])  # align the first pose
+    T_iv_odom_aligned = [T_gt_odom_i @ T_i_vk_odom for T_i_vk_odom in T_iv_odom]
+
+    # middle_idx = np.round(len(T_vi_odom) / 2).astype(int)
+    # T_gt_odom_i = T_iv_gt[middle_idx] @ np.linalg.inv(T_iv_odom[middle_idx])  # align the middle pose
+    # T_iv_odom_aligned = [T_gt_odom_i @ T_i_vk_odom for T_i_vk_odom in T_iv_odom]    
+
+    # Use first 10% of trajectory to compute an alignment transform
```

Comment:
Either delete your commented-out code or add a if statement to select this portion of code when it is supposed to be useful with a function argument to select that behavior.